File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, max = map(int,input().split())
arr = list(map(int,input().split()))
result = []
for i in range(len(arr)):
    for j in range(i+1,len(arr)):
        for q in range(j+1,len(arr)):
            if (arr[i] + arr[j] + arr[q]) <= max:
                result.append(arr[i] + arr[j] + arr[q])
max = 0
for i in range(len(result)):
    if max < result[i]:
        max = result[i]
print(max)

File: 백준/Bronze/2798. 블랙잭/블랙잭.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Scanner;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int number = Integer.parseInt(st.nextToken());
        int maxNumber = Integer.parseInt(st.nextToken());
        int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int total = 0;
        for(int i = 0; i< arr.length-2; i++){
            for(int j = i+1; j<arr.length-1;j++){
                for(int k = j+1; k<arr.length;k++){
                    if (total < arr[i]+arr[j]+arr[k] && arr[i]+arr[j]+arr[k] <=maxNumber){
                        total = arr[i]+arr[j]+arr[k];
                    }
                }
            }
        }
        System.out.println(total);

    }
}
